pkg/provider/aws/config_resources.go

```diff
@@ -1,18 +1,38 @@
 package aws
 
 import (
+	"fmt"
 	"github.com/klothoplatform/klotho/pkg/core"
+	"github.com/klothoplatform/klotho/pkg/provider/aws/resources"
 	"github.com/pkg/errors"
 )
 
-func (a *AWS) GenerateConfigResources(construct *core.Config, result *core.ConstructGraph, dag *core.ResourceGraph) error {
-	if construct.Secret {
-		cfg := a.Config.GetConfig(construct.ID)
-		if cfg.Path == "" {
-			return errors.Errorf("'Path' required for config %s", construct.ID)
-		}
-		return a.generateSecret(construct, result, dag, cfg.Path)
+func (a *AWS) expandConfig(dag *core.ResourceGraph, construct *core.Config) error {
+	if !construct.Secret {
+		return errors.Errorf("unsupported: non-secret config for annotation '%s'", construct.ID)
+	}
+	secretVersion, err := core.CreateResource[*resources.SecretVersion](dag, resources.SecretVersionCreateParams{
+		AppName: a.Config.AppName,
+		Refs:    []core.AnnotationKey{construct.AnnotationKey},
+		Name:    construct.ID,
+	})
+
+	if err != nil {
+		return err
 	}
 
-	return errors.Errorf("unsupported: non-secret config for annotation '%s'", construct.ID)
+	mapped := false
+	for _, r := range dag.GetDownstreamResources(secretVersion) {
+		if secret, ok := r.(*resources.Secret); ok {
+			if err := a.MapResourceToConstruct(secret, construct); err != nil {
+				return err
+			}
+			mapped = true
+			break
+		}
+	}
```

Comment:
cant you call secretVersion.Secret here rather than looking at the downstream resources?

Reply:
Yeah, not sure how I missed that. _I think it was Copilot messing with my IDE's autocomplete._